articles/jwt-auth-angular/index.md

```diff
@@ -0,0 +1,545 @@
+                });
+    }
+}
+```
+
+### Store login token in local storage
+Angular ships with HTTP [interceptors](https://angular.io/api/common/http/HttpInterceptor). We can use this feature to store our login token.  
+
+If the user is logged in, the JWT Interceptor intercepts HTTP requests from the application and attaches a JWT auth token to the Authorization header.  
+
+It's implemented with the HttpInterceptor class from the HttpClientModule; by expanding the HttpInterceptor class, you can build your custom interceptor to alter HTTP requests before they're sent to the server.  
+
+In the provider's section of the app.module.ts file, HTTP interceptors are attached to the request pipeline.  
+
+```ts
+import { Injectable } from '@angular/core';
+import { HttpRequest, HttpHandler, HttpEvent, HttpInterceptor } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import { AuthService } from './service/auth.module';
+
+@Injectable()
+export class JwtInterceptor implements HttpInterceptor {
+    constructor(private authService: AuthService) { }
+
+    intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        // add authorization header with jwt token if available
+        let loggedInUser = this.authService.currentUserValue;
+        if (loggedInUser && loggedInUser.token) {
+            request = request.clone({
+                setHeaders: {
+                    Authorization: `Bearer ${loggedInUser.token}`
+                }
+            });
+        }
+
+        return next.handle(request);
+    }
+}
+```
+
+Now let's go ahead and add this script in our `app.module.ts`.  
+
+```ts
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { ReactiveFormsModule } from '@angular/forms';
+import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
+
+import { AppComponent } from './app.component';
+import { appRoutingModule } from './app.routing';
+
+import { tokenInterceptor} from 'helpers/tokenInterceptor';
+import { DashboardComponent } from './dashboard';
+import { LoginComponent } from './login';
+
+@NgModule({
+    imports: [
+        BrowserModule,
+        ReactiveFormsModule,
+        HttpClientModule,
+        appRoutingModule
+    ],
+    declarations: [
+        AppComponent,
+        HomeComponent,
+        LoginComponent
+    ],
+    providers: [
+        { provide: HTTP_INTERCEPTORS, useClass: tokenInterceptor, multi: true 
+    ],
+    bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+Well, let's start our angular application by running the following command:  
+
+```bash
+ng serve --open //starts on port 4200 by default unless you specfied otherwise
+```
+You can now make requests to our PHP endpoint and login while the generated token is stored in your browser's local storage.  
+
+### Conclusion
+You learned how to use JWT authentication in your Angular 11 framework in this tutorial.  
```

Comment:
Consider changing "You learned how to use JWT authentication in your Angular 11 framework in this tutorial." to "You have learned how to use JWT authentication in your Angular 11 application in this tutorial. Implement other authentication strategies such as token authentication in your Angular app."